src/Plotting/StrengthEnvelope.jl

```diff
@@ -0,0 +1,319 @@
+    Ttop = 0
+    Tbot = 1350
+end
+
+function GP_Compute_ThermalStructure(Z, s::LinearTemp)
+    @unpack Ttop, Tbot  = s
+
+    Temp  = zeros(Float64, length(Z))
+
+    dz    = Z[end]-Z[1];
+    dT    = Tbot - Ttop
+
+    Temp .= abs.(Z./dz).*dT .+ Ttop
+    return Temp
+end
+
+"""
+    HalfspaceCoolingTemp(Tsurface=0, Tmantle=1350, Age=60, Adiabat=0)
+    
+Sets a halfspace temperature structure in plate
+Parameters
+========
+- Tsurface : surface temperature [C]
+- Tmantle : mantle temperature [C]
+- Age : Thermal Age of plate [Myrs]
+- Adiabat : Mantle Adiabat [K/km]
+"""
+@with_kw_noshow mutable struct HalfspaceCoolingTemp <: AbstractThermalStructure
+    Tsurface = 0        # top T
+    Tmantle  = 1350     # bottom T
+    Age      = 60       # thermal age of plate [in Myrs]
+    Adiabat  = 0        # Adiabatic gradient in K/km
+    kappa    = 1e-6     # thermal diffusivity
+end
+
+function GP_Compute_ThermalStructure(Z, s::HalfspaceCoolingTemp)
+    @unpack Tsurface, Tmantle, Age, Adiabat, kappa  = s
+
+    Temp  = zeros(Float64, length(Z))
+
+    MantleAdiabaticT    =   Tmantle .+ Adiabat*abs.(Z);   # Adiabatic temperature of mantle
+    
+    for i in eachindex(Temp)
+        Temp[i] =   (Tsurface .- Tmantle)*erfc((abs.(Z[i]))./(2*sqrt(kappa*Age))) + MantleAdiabaticT[i];
+    end
+    return Temp
+end
+
+"""
+    LithPres(MatParam, Phases, ρ, T, nz, dz, g)
+
+Iteratively solves a 1D lithostatic pressure profile (compatible with temperature- and pressure-dependent densities)
+Parameters
+========
+- ρ:  density vector for initial guess(can be zeros)
+- T:  temperature vector
+- dz: grid spacing
+- g:  gravitational accelaration
+"""
+function LithPres(MatParam, Phases, ρ, T, dz, g)
+    nz   = length(T)
+    P    = zeros(Float64, nz)
+
+    tol  = 1e-6
+    n    = 0
+    stop = 1
+    iter = 0
+    
+    while stop > tol && iter < 100
+        # store current norm
+        nprev = n
+        iter += 1
+
+        # update densities
+        args  = (T=T, P=P)
+        compute_density!(ρ, MatParam, Phases, args)
+
+        # update pressure
+        for i = 2 : nz
+            P[i] = P[i-1] + ρ[i-1] * g * dz
+        end
+
+        # compute new norm
+        n    = (sum(P.^2)).^0.5
+
+        # evaluate stopping criterium
+        stop = abs((n-nprev) / (n+nprev))
+    end
+
+    #println("Finding lithostatic pressure took $iter iterations.")
+    return P
+end
+
+function solveStress(MatParam, Phases, ε, P, T)
+    # solve for stress
+    nz = length(T)
+    τ  = zeros(Float64, nz)
+    for i = 1 : nz
+        Pres = P[i]
+        Temp = T[i]
+        args = (T=Temp, P=Pres)
+        Mat  = MatParam[Phases[i]]
+        τ[i] = compute_τII(Mat.CreepLaws[1], ε, args)
+
+        F    = compute_yieldfunction(Mat.Plasticity[1], P=Pres, τII=τ[i])
+        if F > 0
+            ϕ = Mat.Plasticity[1].ϕ.val
+            c = Mat.Plasticity[1].C.val
+            τ[i] = Pres * sind(ϕ) + cosd(ϕ) * c
```

Comment:
`sind(ϕ)` and `cosd(ϕ)` are already inside the Drucker-Prager struct. They are called `Mat.Plasticity[1].sinϕ` and  `Mat.Plasticity[1].cosϕ`

Reply:
> `sind(ϕ)` and `cosd(ϕ)` are already inside the Drucker-Prager struct. They are called `Mat.Plasticity[1].sinϕ` and `Mat.Plasticity[1].cosϕ`

Thanks for all the suggestions. Implemented all of the above and checked functionality.